src/emulator/controller.ts

```diff
@@ -528,42 +527,7 @@ export async function startAll(options: any, noUi: boolean = false): Promise<voi
       for (const f of files) {
         const fPath = path.join(databaseExportDir, f);
         const ns = path.basename(f, ".json");
-
-        databaseLogger.logLabeled("BULLET", "database", `Importing data from ${fPath}`);
-
-        const readStream = fs.createReadStream(fPath);
-
-        await new Promise((resolve, reject) => {
-          const req = http.request(
-            {
-              method: "PUT",
-              host: databaseAddr.host,
-              port: databaseAddr.port,
-              path: `/.json?ns=${ns}&disableTriggers=true&writeSizeLimit=unlimited`,
-              headers: {
-                Authorization: "Bearer owner",
-                "Content-Type": "application/json",
-              },
-            },
-            (response) => {
-              if (response.statusCode === 200) {
-                resolve();
-              } else {
-                databaseLogger.log("DEBUG", "Database import failed: " + response.statusCode);
-                response
-                  .on("data", (d) => {
-                    databaseLogger.log("DEBUG", d.toString());
-                  })
-                  .on("end", reject);
-              }
-            }
-          );
-
-          req.on("error", reject);
-          readStream.pipe(req, { end: true });
-        }).catch((e) => {
-          throw new FirebaseError("Error during database import.", { original: e, exit: 1 });
-        });
+        await databaseEmulator.importData(ns, fPath);
```

Comment:
This logic is moved mostly  unchanged to be inside the emulator.  Only change is to add successful imports to the internal state.